src/Servers/IIS/AspNetCoreModuleV2/OutOfProcessRequestHandler/serverprocess.cpp

```diff
@@ -1118,6 +1125,52 @@ SERVER_PROCESS::SetupStdHandles(
     return hr;
 }
 
+
+void
+SERVER_PROCESS::ReadStdErrHandle(
+    LPVOID pContext
+)
+{
+    auto pLoggingProvider = static_cast<SERVER_PROCESS*>(pContext);
+    DBG_ASSERT(pLoggingProvider != NULL);
+    pLoggingProvider->ReadStdErrHandleInternal();
+}
+
+void
+SERVER_PROCESS::ReadStdErrHandleInternal()
+{
+    std::string tempBuffer;
+    tempBuffer.resize(4096);
+    size_t m_charactersLeft = 30000;
+
+    // If ReadFile ever returns false, exit the thread
+    DWORD dwNumBytesRead = 0;
+    while (true)
+    {
+        if (ReadFile(m_hStdoutHandle,
+            tempBuffer.data(),
+            4096,
+            &dwNumBytesRead,
+            nullptr))
+        {
+            auto text = to_wide_string(tempBuffer.substr(0, dwNumBytesRead), GetConsoleOutputCP());
+            auto const writeSize = min(m_charactersLeft, text.size());
+            m_output.write(text.c_str(), writeSize);
+            m_charactersLeft -= writeSize;
+        }
+        else
+        {
+            auto hr = GetLastError();
```

Comment:
Should something be done here? Otherwise might as well just return without checking last error.

Reply:
I think we shouldn't handle errors here. There are at least 3 different errors that could be returned that are false positives. We also cancel reading once the process exits.